src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/NuGetProjectFactory.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Shell;
+using NuGet.ProjectManagement;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Composite provider chaining calls to other providers.
+    /// </summary>
+    public class NuGetProjectFactory : IProjectSystemProvider
```

Comment:
let's have a different (or no) interface, then this type can import all on IProjectSystemProvider